bevy_lint/src/lints/duplicate_bevy_dependencies.rs

```diff
@@ -0,0 +1,38 @@
+//! Checks for multiple versions of `bevy` in the dependencies.
+//!
+//! # Motivation
+//!
+//! When different third party crates use incompatible versions of Bevy, it can lead to confusing
+//! errors and type incompatibilities.
+
+use crate::declare_bevy_lint;
+use clippy_utils::{diagnostics::span_lint, find_crates};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::declare_lint_pass;
+use rustc_span::Symbol;
+
+declare_bevy_lint! {
+    pub DUPLICATE_BEVY_DEPENDENCIES,
+    CORRECTNESS,
+    "duplicate bevy dependencies",
+}
+
+declare_lint_pass! {
+     DuplicateBevyDependencies => [DUPLICATE_BEVY_DEPENDENCIES.lint]
+}
+
+impl<'tcx> LateLintPass<'tcx> for DuplicateBevyDependencies {
+    fn check_crate(&mut self, cx: &LateContext<'tcx>) {
+        let bevy_crates = find_crates(cx.tcx, Symbol::intern("bevy"));
```

Comment:
(Nit) Calling `Symbol::intern()` / `sym!()` is an expensive operation that should probably be cached. I solve this by storing the `Symbol` as a field in the lint pass in #184.

```rust
// Sketch of what you may do...
struct DuplicateBevyDependencies {
    bevy_symbol: Symbol,
}

impl Default for DuplicateBevyDependencies {
    fn default() -> Self {
        Self { bevy_symbol: sym!(bevy) }
    }
}

// This is different from `declare_lint_pass!`, since it does not actually define the lint pass structure.
impl_lint_pass! {
    DuplicateBevyDependencies => [DUPLICATE_BEVY_DEPENDENCIES.lint]
}
```